prov/gni/src/gnix_rma.c

```diff
@@ -191,16 +237,12 @@ static int __gnix_rma_send_data_req(void *arg)
 	int rc;
 	int inject_err = _gnix_req_inject_err(req);
 
-	txd = (struct gnix_tx_descriptor *)req->txd;
-	if (!txd) {
-		rc = _gnix_nic_tx_alloc(nic, &txd);
-		if (rc) {
-			GNIX_INFO(FI_LOG_EP_DATA,
-				  "_gnix_nic_tx_alloc() failed: %d\n",
-				  rc);
-			return -FI_ENOSPC;
-		}
-		req->txd = txd;
+	rc = _gnix_nic_tx_alloc(nic, &txd);
+	if (rc) {
+		GNIX_INFO(FI_LOG_EP_DATA,
+				"_gnix_nic_tx_alloc() failed: %d\n",
+				rc);
+		return -FI_ENOSPC;
```

Comment:
As explained above, the req->txd field isn't used in this file any more.  I am going towards consistently allocating a TXD when a request is initiated and freeing it as soon as it's completed.